mod/messageboard/actions/messageboard/add.php

```diff
@@ -6,33 +6,26 @@
  */
 
 $message_content = get_input('message_content');
-$owner_guid = get_input("owner_guid");
-$owner = get_entity($owner_guid);
+$owner_guid = (int) get_input('owner_guid');
+$owner = get_user($owner_guid);
 
-if ($owner && !empty($message_content)) {
-	$result = messageboard_add(elgg_get_logged_in_user_entity(), $owner, $message_content, $owner->access_id);
-
-	if ($result) {
-		system_message(elgg_echo("messageboard:posted"));
-
-		$options = array(
-			'annotations_name' => 'messageboard',
-			'guid' => $owner->getGUID(),
-			'limit' => $num_display,
-			'pagination' => false,
-			'reverse_order_by' => true,
-			'limit' => 1
-		);
-
-		$output = elgg_list_annotations($options);
-		echo $output;
+if (!$owner || empty($message_content)) {
+	return elgg_error_response(elgg_echo('messageboard:blank'));
+}
 
-	} else {
-		register_error(elgg_echo("messageboard:failure"));
-	}
+$result = messageboard_add(elgg_get_logged_in_user_entity(), $owner, $message_content, $owner->access_id);
 
-} else {
-	register_error(elgg_echo("messageboard:blank"));
+if (!$result) {
+	return elgg_error_response(elgg_echo('messageboard:failure'));
 }
 
-forward(REFERER);
+$output = elgg_list_annotations([
```

Comment:
Hmm. Not how I intrnded to use this. I'd rather pass annotation instance and a view if needed. The response factory will export an annotation into an object.

```php
[
   'annotation' => ElggAnnotation,
   'view' => string,
   'action' => 'create',
]
```